### ISSUE TITLE
Cannot deserialize response data which contains values of type None

### ISSUE THREAD

**Issue by tjprescott** [Azure/azure-sdk-for-python#544]
When trying to show a resource:
```
resource_client.resources.get(
            resource_group_name=<group_name>
            resource_name=<name>,
            resource_provider_namespace=<namespace>,
            resource_type=<type>,
            api_version=<api_version>,
            parent_resource_path=<path> or ''
        )
```
I get an msrest DeserializationError:

> Unable to deserialize response data. Data: {...}, object, TypeError: Cannot deserialize generic object with type: `<class 'NoneType'>`

The response data contains the information I want although numerous fields have None as their value.

**Comment by lmazuel** [comment]
I'm not able to reproduce your problem. If you were able to get the JSON answer, could you add it in this issue?
Thank you

**Comment by tjprescott** [comment]
This is the exception message with the data:
> Unable to deserialize response data. Data: {'siteMode': None, 'resourceGroup': 'FooResourceGroup', 'perSiteScaling': False, 'geoRegion': 'West US', 'currentWorkerSize': 0, 'hostingEnvironmentProfile': None, 'webSpace': 'FooWebspace', 'planName': 'VirtualDedicatedPlan', 'workerSizeId': 0, 'workerSize': 0, 'serverFarmId': 0, 'maximumNumberOfWorkers': 1, 'workerTierName': None, 'subscription': 'FooSubscriptionId', 'numberOfSites': 0, 'hostingEnvironmentId': None, 'adminSiteName': None, 'adminRuntimeSiteName': None, 'tags': None, 'computeMode': 0, 'kind': None, 'numberOfWorkers': 0, 'name': 'FooResourceName', 'currentNumberOfWorkers': 0, 'status': 0, 'hostingEnvironment': None, 'currentWorkerSizeId': 0}, object, TypeError: Cannot deserialize generic object with type: <class 'NoneType'>

And here's the call stack upstream of our code
>   File "C:\Users\t\Documents\github\azure-cli\env\lib\site-packages\azure\mgmt\resource\resources\operations\resources_operations.py", line 462, in get
>     deserialized = self._deserialize('GenericResource', response)
>   File "C:\Users\t\Documents\github\azure-cli\env\lib\site-packages\msrest\serialization.py", line 623, in __call__
>     value = self.deserialize_data(raw_value, attr_type)
>   File "C:\Users\t\Documents\github\azure-cli\env\lib\site-packages\msrest\serialization.py", line 736, in deserialize_data
>     raise_with_traceback(DeserializationError, msg, err)
>   File "C:\Users\t\Documents\github\azure-cli\env\lib\site-packages\msrest\exceptions.py", line 47, in raise_with_traceback
>     raise error.with_traceback(exc_traceback)
>   File "C:\Users\t\Documents\github\azure-cli\env\lib\site-packages\msrest\serialization.py", line 722, in deserialize_data
>     data_val = self.deserialize_type[data_type](data)
>   File "C:\Users\t\Documents\github\azure-cli\env\lib\site-packages\msrest\serialization.py", line 784, in deserialize_object
>     value, **kwargs)
>   File "C:\Users\t\Documents\github\azure-cli\env\lib\site-packages\msrest\serialization.py", line 801, in deserialize_object
>     raise TypeError(error + str(obj_type))

**Comment by tjprescott** [comment]
So, when I use the SDK method to retrieve a storage account, it worked fine. The above issue occurred when I tried to retrieve a server farm resource.

**Comment by lmazuel** [comment]
I think you fall into this known bug https://github.com/Azure/azure-rest-api-specs/issues/137
However, I didn't open that bug with the exact same code as you (I used the mgmt-web API directly)
I will try asap to confirm this is the same bug.

**Comment by lmazuel** [comment]
I was wrong, it was not related. I just created a PR to the underlying runtime library for fix.

**Comment by lmazuel** [comment]
Hi,

I just released msrest 0.1.3 which fixes this bug
https://pypi.python.org/pypi/msrest/0.1.3

Thank you for your report again!
